src/Compilers/Core/Portable/DocumentationCommentId.cs

```diff
@@ -1484,12 +1484,12 @@ private static string ParseName(string id, ref int index)
                 int delimiterOffset = id.IndexOfAny(s_nameDelimiters, index);
                 if (delimiterOffset >= 0)
                 {
-                    name = id.Substring(index, delimiterOffset - index);
+                    name = id[index..delimiterOffset];
```

Comment:
>name = id[index..delimiterOffset]; [](http://example.com/codeflow?start=20&length=34)

This is a style only change. Please revert. BTW, it looks like the change results in worse IL.

Reply:
Can we make it not result in worse IL? Because it's certainly more readable.

Reply:
> Can we make it not result in worse IL? 

Even if we could do that in the future (feel free to open an issue). We do not make change for the sake of changes. Style only changes are not welcome.

> Because it's certainly more readable.

That is an opinion, not a fact. We prefer to not argue about personal stylistic preferences.

Reply:
it's def more readable, afaict, objectively so.  One needs to do all the math in their head to figure out that its' the same as what a range/index clearly demonstrates.

Runtime code seems to be good as well.  Measuring this out (even on netfx doesn't seem to indicate any sort of problem here).

Reply:
This is not style-only.  It is conveyance of meaning and intent.  Magic numbers versus clear concepts is not a style.